misc/SW_WA_2021_2024/guilderton_to_geraldton/swals/README.md

```diff
@@ -0,0 +1,390 @@
+# can be found with:
+grep -B1 stable *.log 
+# To focus just on the global domains, which have domain index 1 (in this case):
+grep -B1 stable *.log | grep "0000001 ,"
+# If the minimum global domain timestep is much larger than the chosen global_dt, you might
+# consider increasing global_dt. However that will fatten the halo regions, so there is
+# a trade-off. You can test by comparing "load balanced models" with alternative global_dt.
+
+#
+# RUN A LOAD BALANCED MODEL
+#
+
+# If everything looks sensible then make a load balance file from within the
+# multidomain directory
+Rscript ../../../post_process/load_balance_script.R
+# This will print information on the expected improvement.
+
+# If the speed up looks good, then store the file and run a new load balanced model.
+mkdir -p ../../../load_balance_files
+cp load_balance_partition.txt ../../../load_balance_files/load_balance_16MPI_NNL4_defaultres.txt
+# Go back to the "swals" directory
+cd ../../../
+# Make a new model configuration namelist that is similar to the previous one, but uses the newly 
+# created load balance file
+cp multidomain_design_control_NNL4_defaultres_defaultloadbalance.nml multidomain_design_control_NNL4_defaultres.nml 
+code multidomain_design_control_NNL4_defaultres.nml # (EDIT the load balance file)
+# Make a run script that uses the new load balance file.
+code run/test_model_with_load_balance_sapphirerapids.pbs # (EDIT to use the new multidomain_design....nml)
+# Run a short load balanced model and check that the speedup is OK. 
+qsub run/test_model_with_load_balance_sapphirerapids.pbs
+
+# If the speedup isn't good, or a good speedup isn't possible according to the load_balance_script.R, then
+# go to the multidomain output directory and check the time spent in "evolve:".
+cd name_of_multidomain_folder
+grep " evolve:" *.log
+# (Ignore the overall times which can be strongly affected by the variable disk speed when reading inputs).
+# Compare this to the time in parallel communication -- ideally the latter
+# should be nearly equal between processes, and small compared to the evolve
+# time.
+grep "comms1" *.log
+# Compute the average of these times using: 
+grep " comms1:" *.log | awk 'NF>1{print $(NF-1)}' | awk '{sum += $1; count++} END {if (count > 0) print sum/count}'
+# If the communication is quite unequal between processes, check the model for
+# individual domains with long run times (comparable to the evolve time you can
+# achieve) which can prevent load balancing.
+Rscript ../../../post_process/report_domain_runtimes.R
+# Any problematic domains might be split or coarsened or removed, to reduce their run time and
+# allow balancing the model.
+
+# If the time in "comms1" is quite equal, but seems too large, consider running with fewer MPI
+# processes. [Or, redesigning the model to reduce the halo fatness].
+
+
+#
+# Below here I assume the model design matches the intent, and it is reasonably efficient.
```

Comment:
Somewhere you might add a comment "Go back to the swals folder `../../../`" since the previous commands are in a model run folder.